cmd/kueuectl/app/list/list_pods.go

```diff
@@ -0,0 +1,394 @@
+)
+
+const (
+	podLong = `Lists all pods that matches the given criteria: Should be part of the specified Job kind,
+belonging to the specified namespace, matching
+the label selector or the field selector.`
+	podExample = `  # List Pods
+kueuectl list pods --for job/job-name`
+)
+
+var jobControllersWithPodLabelSelector = []jobControllerWithPodLabelSelector{
+	&kueuejob.Job{},
+	&kueuejobset.JobSet{},
+	&kueuemxjob.JobControl{},
+	&kueuepaddlejob.JobControl{},
+	&kueuetfjob.JobControl{},
+	&kueuepytorchjob.JobControl{},
+	&kueuexgboostjob.JobControl{},
+	&kueuempijob.MPIJob{},
+	&pod.Pod{},
+	&kueueraycluster.RayCluster{},
+	&kueuerayjob.RayJob{},
+}
+
+type PodOptions struct {
+	PrintFlags *genericclioptions.PrintFlags
+
+	Limit                  int64
+	AllNamespaces          bool
+	ServerPrint            bool
+	Namespace              string
+	LabelSelector          string
+	FieldSelector          string
+	UserSpecifiedForObject string
+	ForName                string
+	ForGVK                 schema.GroupVersionKind
+	ForObject              *unstructured.Unstructured
+	PodLabelSelector       string
+
+	Clientset k8s.Interface
+
+	genericiooptions.IOStreams
+}
+
+type jobControllerWithPodLabelSelector interface {
+	Object() client.Object
+	GVK() schema.GroupVersionKind
+	PodLabelSelector() string
+}
+
+func NewPodOptions(streams genericiooptions.IOStreams) *PodOptions {
+	return &PodOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
+		IOStreams:  streams,
+	}
+}
+
+func NewPodCmd(clientGetter util.ClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewPodOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:                   "pods --for TYPE[.API-GROUP]/NAME",
+		DisableFlagsInUseLine: true,
+		Aliases:               []string{"po"},
+		Short:                 "List Pods belong to a Job Kind",
+		Long:                  podLong,
+		Example:               podExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			err := o.Complete(cmd, clientGetter)
+			if err != nil {
+				return err
+			}
+			if o.ForObject == nil {
+				return nil
+			}
+			if len(o.PodLabelSelector) == 0 {
+				return nil
+			}
+			return o.Run(cmd.Context(), clientGetter)
+		},
+	}
+
+	o.PrintFlags.AddFlags(cmd)
+
+	addAllNamespacesFlagVar(cmd, &o.AllNamespaces)
+	addFieldSelectorFlagVar(cmd, &o.FieldSelector)
+	addLabelSelectorFlagVar(cmd, &o.LabelSelector)
+	addForObjectFlagVar(cmd, &o.UserSpecifiedForObject)
+
+	_ = cmd.MarkFlagRequired("for")
+
+	return cmd
+}
+
+// Complete takes the command arguments and infers any remaining options.
+func (o *PodOptions) Complete(cmd *cobra.Command, clientGetter util.ClientGetter) error {
+	var err error
+
+	o.Limit, err = listRequestLimit()
+	if err != nil {
+		return err
+	}
+
+	outputOption := ptr.Deref(o.PrintFlags.OutputFormat, "")
+	if outputOption == "" || strings.Contains(outputOption, "wide") {
```

Comment:
why only in these 2 cases?

Also... it doesn't look like it's server-side.... it's just kubectl code, as opposed to our code. But, it still happens in the client, as far as I can tell.

Reply:
> why only in these 2 cases?

Another cases "name", "yaml", "json" we can show on client side.

Reply:
> Also... it doesn't look like it's server-side.... it's just kubectl code, as opposed to our code. But, it still happens in the client, as far as I can tell.

Yes, but the difference is that the table printer prints the already calculated result from the server part.

Reply:
To be more clear if we adding header `Accept=application/json;as=Table;v=%s;g=%s` (look [here](https://github.com/kubernetes-sigs/kueue/blob/dcd5bc8e68d5dd6010d3121308b4c75536fc822b/cmd/kueuectl/app/list/list_pods.go#L381-L384)) backend returning already calculated table from backend instead of full object.

Reply:
I see, thanks.